sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala

```diff
@@ -346,6 +346,33 @@ object RapidsPluginUtils extends Logging {
       loadExtensions(classOf[SparkPlugin], pluginClasses)
     }
   }
+
+  /**
+   * Internal implementation to extract supported GPU architectures from properties file
+   * and check if current GPU architecture is supported.
+   */
+  private def checkGpuArchitectureInternal(propName: String): Unit = {
+    val props = RapidsPluginUtils.loadProps(propName)
```

Comment:
probably not a big deal yet, but our code already has the issue of rescaning the classpath/reparse the same properties  resource unless the result is memoized internally by the JDK

Reply:
I was thinking the same thing! :smile: However it might not be so redundant here as it appears.  I think most of the props loading occurs in the driver plugin, but this is in the executor plugin.

Reply:
Excellent point